### PR TITLE
VODE optimizations

### PR DESCRIPTION
**Pull request by zingale**
Eliminate the storage of the old state in rpar
No longer initialize ydot to 0 in the wrappers.

This addresses parts of #251

### CODE REVIEW

**Comment by zingale** [comment]
all tests pass on this branch

File: integration/VODE/vode_type.F90 [hunk 1]
```diff
@@ -266,8 +264,10 @@ subroutine burn_to_vode(state, y, rpar)
     rpar(irp_dx)                             = state % dx
 
     rpar(irp_Told)                           = state % T_old
-    rpar(irp_dcvdt)                          = state % dcvdt
-    rpar(irp_dcpdt)                          = state % dcpdt
+
+    ! we don't need to update the specific heats, since these are not
```
Comment (maxpkatz) [inline integration/VODE/vode_type.F90]: While this comment is true, the actual network doesn't modify any of these other variables either. In fact, why do we even do burn_to_vode?

Comment (zingale) [reply integration/VODE/vode_type.F90]: yeah, that's something I haven't figured out.

Comment (zingale) [reply integration/VODE/vode_type.F90]: indeed, it doesn't seem to be needed.  I'm going to remove burn_to_vode completely.

**Comment by zingale** [comment]
test output:
http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2020-03-08-001/index.html

**Comment by dwillcox** [comment]
http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2020-03-11-005/index.html

This PR looks ready to merge :-)

**Comment by zingale** [comment]
all tests pass with the last commits

**Pull request merged**